### PR TITLE
chore(schemas): update to pydantic v2

### LINKED ISSUE #11214: Update schemas pydantic to v2
We're on a now-quite-old version of pydantic, and some external dependencies on the data side are requesting that we update

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/EXP-4755)

### PR DESCRIPTION
**Pull request by mikewilli**
Because

- the schemas package is used by a lot of data tooling
- the data tooling is starting to complain about schemas requiring pydantic v1 due to other dependencies
- the pydantic-to-typescript package is outdated and does not support pydantic v2

This commit

- updates the schemas package to v2
- changes the typescript generation to a script internal to experimenter

Fixes #11214

### CODE REVIEW

File: schemas/generate_json_schema.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
+"""
```
Comment (jaredlockhart) [inline schemas/generate_json_schema.py:1]: Eeeeehhhhhh I wish we didn't have to do this.  There's not something else that works off the shelf?

Comment (mikewilli) [reply schemas/generate_json_schema.py:1]: So yea, I agree. I think there may be some options, but nothing that jumped out at me when doing some initial searching. I also know Beth was interested in pursuing something like Zod and I don't really have the time to evaluate things like that at the moment. This is becoming more of an issue for data so I saw this as a bit of a stopgap, but taking inspiration from tooling we already use.

Comment (jaredlockhart) [reply schemas/generate_json_schema.py:1]: Okey doke fair enough.  Let's do it for now to unblock things and maybe we can find a better solution in the future.  Thanks for taking this on @mikewilli 🙏

**Review by jaredlockhart — APPROVED** [review]
Awesome ty @mikewilli 🙏

**Pull request merged**